client/handle_http_test.go

```diff
@@ -908,111 +891,72 @@ func TestRecursiveUploadsAndDownloads(t *testing.T) {
 	tempFile1, err := os.CreateTemp(tempDir, "test1")
 	assert.NoError(t, err, "Error creating temp1 file")
 	tempFile2, err := os.CreateTemp(tempDir, "test1")
 	assert.NoError(t, err, "Error creating temp2 file")
+	innerTempFile, err := os.CreateTemp(innerTempDir, "testInner")
 	defer os.Remove(tempFile1.Name())
 	defer os.Remove(tempFile2.Name())
+	defer os.Remove(innerTempFile.Name())
+
 	_, err = tempFile1.WriteString(testFileContent1)
 	assert.NoError(t, err, "Error writing to temp1 file")
 	tempFile1.Close()
 	_, err = tempFile2.WriteString(testFileContent2)
 	assert.NoError(t, err, "Error writing to temp2 file")
 	tempFile2.Close()
+	_, err = innerTempFile.WriteString(innerTestFileContent)
+	assert.NoError(t, err, "Error writing to inner test file")
+	innerTempFile.Close()
 
 	t.Run("testPelicanRecursiveGetAndPutOsdfURL", func(t *testing.T) {
-		config.SetPreferredPrefix("pelican")
+		config.SetPreferredPrefix("PELICAN")
 		// Set path for object to upload/download
 		tempPath := tempDir
 		dirName := filepath.Base(tempPath)
-		// Note: minimally fixing this test as it is soon to be replaced
-		uploadURL := "pelican://" + param.Server_Hostname.GetString() + ":" + strconv.Itoa(param.Server_WebPort.GetInt()) + "/test/" + dirName
-
-		//////////////////////////////////////////////////////////
-
-		// Upload the file with PUT
-		transferDetailsUpload, err := DoPut(ctx, tempDir, uploadURL, true, WithTokenLocation(tempToken.Name()))
-		require.NoError(t, err)
-		if err == nil && len(transferDetailsUpload) == 2 {
-			countBytes17 := 0
-			countBytes23 := 0
-			// Verify we got the correct files back (have to do this since files upload in different orders at times)
-			for _, transfer := range transferDetailsUpload {
-				transferredBytes := transfer.TransferredBytes
-				switch transferredBytes {
-				case int64(17):
-					countBytes17++
-					continue
-				case int64(23):
-					countBytes23++
-					continue
-				default:
-					// We got a byte amount we are not expecting
-					t.Fatal("did not upload proper amount of bytes")
-				}
-			}
-			if countBytes17 != 1 || countBytes23 != 1 {
-				// We would hit this case if 1 counter got hit twice for some reason
-				t.Fatal("One of the files was not uploaded correctly")
-			}
-		} else if len(transferDetailsUpload) != 2 {
-			t.Fatalf("Amount of transfers results returned for upload was not correct. Transfer details returned: %d", len(transferDetailsUpload))
-		}
+		uploadStr := "osdf:///test/" + dirName
+		uploadURL, err := url.Parse(uploadStr)
+		assert.NoError(t, err)
 
-		// Download the files we just uploaded
-		transferDetailsDownload, err := DoGet(ctx, uploadURL, t.TempDir(), true, WithTokenLocation(tempToken.Name()))
+		// For OSDF url's, we don't want to rely on osdf metadata to be running therefore, just ensure we get correct metadata for the url:
+		pelicanURL, err := newPelicanURL(uploadURL, "osdf")
 		assert.NoError(t, err)
-		if err == nil && len(transferDetailsUpload) == 2 {
```

Comment:
I can't see this because it is on an older commit. I know I change the tests a bit with a later commit so just let me know if it still persists after I commit these fixes.